aviatrix/resource_aviatrix_cloudn_registration.go

```diff
@@ -0,0 +1,222 @@
+		Password:          client.Password,
+	}
+
+	cloudnClient, err := goaviatrix.NewClient(d.Get("username").(string), d.Get("password").(string), d.Get("address").(string), nil)
+	if err != nil {
+		return diag.Errorf("failed to initialize Aviatrix CloudN Client: %v", err)
+	}
+
+	d.SetId(cloudnRegistration.Name)
+	flag := false
+	defer resourceAviatrixCloudnRegistrationReadIfRequired(ctx, d, meta, &flag)
+
+	err = cloudnClient.CreateCloudnRegistration(ctx, cloudnRegistration)
+	if err != nil {
+		return diag.Errorf("failed to create Aviatrix CloudN Registration: %v", err)
+	}
+
+	gateway := &goaviatrix.TransitVpc{
+		GwName: cloudnRegistration.Name,
+	}
+	if _, ok := d.GetOk("local_as_number"); ok {
+		localASNumber := d.Get("local_as_number").(string)
+		err := client.SetLocalASNumber(gateway, localASNumber)
+		if err != nil {
+			return diag.Errorf("failed to create Aviatrix CloudN Registration: could not set local_as_number: %v", err)
+		}
+	}
+
+	if _, ok := d.GetOk("prepend_as_path"); ok {
+		var prependASPath []string
+		for _, v := range d.Get("prepend_as_path").([]interface{}) {
+			prependASPath = append(prependASPath, v.(string))
+		}
+
+		err := client.SetPrependASPath(gateway, prependASPath)
+		if err != nil {
+			return diag.Errorf("failed to create Aviatrix CloudN Registration: could not set prepend_as_path: %v", err)
+		}
+	}
+
+	return resourceAviatrixCloudnRegistrationReadIfRequired(ctx, d, meta, &flag)
+}
+
+func resourceAviatrixCloudnRegistrationReadIfRequired(ctx context.Context, d *schema.ResourceData, meta interface{}, flag *bool) diag.Diagnostics {
+	if !(*flag) {
+		*flag = true
+		return resourceAviatrixCloudnRegistrationRead(ctx, d, meta)
+	}
+	return nil
+}
+
+func resourceAviatrixCloudnRegistrationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*goaviatrix.Client)
+
+	if d.Get("name").(string) == "" {
+		id := d.Id()
+		log.Printf("[DEBUG] Looks like an import, no name received. Import Id is %s", id)
+		d.Set("name", id)
+		d.SetId(id)
+	}
+
+	cloudnRegistration := &goaviatrix.CloudnRegistration{
+		Name: d.Get("name").(string),
+	}
+	cloudnRegistration, err := client.GetCloudnRegistration(ctx, cloudnRegistration)
+	if err != nil {
+		if err == goaviatrix.ErrNotFound {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("failed to read Aviatrix CloudN Registration: %v", err)
+	}
+
+	d.Set("address", cloudnRegistration.ControllerAddress)
+
+	gateway := &goaviatrix.TransitVpc{
+		GwName: d.Get("name").(string),
+	}
+	transitGatewayAdvancedConfig, err := client.GetTransitGatewayAdvancedConfig(gateway)
+	if err != nil {
+		return diag.Errorf("failed to read Aviatrix Cloudn Registration transit gateway advanced config: %v", err)
+	}
+	if transitGatewayAdvancedConfig.LocalASNumber != "" {
+		d.Set("local_as_number", transitGatewayAdvancedConfig.LocalASNumber)
+		d.Set("prepend_as_path", transitGatewayAdvancedConfig.PrependASPath)
+	}
```

Comment:
prepend_as_path can only be set after local_as_number has been set because prepend_as_path can only be an array of repeated local_as_numbers. It is possible to set local_as_number without setting prepend_as_path, but not the other way around

Reply:
I fixed create logic to so that prepeand_as_path is only accepted if local_as_number is given.